cpp/src/arrow/filesystem/azurefs.cc

```diff
@@ -1728,6 +1728,37 @@ class AzureFileSystem::Impl {
     }
     return Status::OK();
   }
+
+  Status DeleteFile(const AzureLocation& location) {
+    if (location.path.empty()) {
+      return Status::Invalid("Cannot delete an empty path");
+    }
+
+    auto last_sep = location.path.find_last_of(internal::kSep);
+    auto directory = location.path.substr(0, last_sep);
+    auto file = location.path.substr(last_sep + 1);
+
+    auto file_client =
+      datalake_service_client_->GetFileSystemClient(location.container)
+        .GetDirectoryClient(directory).GetFileClient(file);
+
+    try {
+      auto response = file_client.DeleteIfExists();
+      if (response.Value.Deleted) {
+        return Status::OK();
+      } else {
+        return StatusFromErrorResponse(
+            container_client.GetUrl(), response.RawResponse.get(),
+            "Failed to delete a container: " + location.container);
+      }
+    } catch (const Azure::Storage::StorageException& exception) {
+      return internal::ExceptionToStatus(
+          "Failed to delete a file: " + location.all + ": " +
+              file_client.GetUrl(),
+          exception);
```

Comment:
It's been a while since you rebased. This function expects exception to be the first parameter now and all the other components can be passed as separate arguments, so you don't have to `+` the strings yourself.